### ISSUE TITLE
Broken Yarn Projects due to crash inside the compiler

### ISSUE THREAD

**Issue by dukiodev** [YarnSpinnerTool/YarnSpinner-Unity#182]
**What is the current behavior?**
Yarn Project seems to break (broken link with the original script and missing thumbnail) whenever I try to modify the attributes of the Yarn Project, like switching language or adding another one to the list of localisations. After apply the new settings, the project breaks if there were any instances of the "visited" bult-in function.

**Please provide the steps to reproduce, and if possible a minimal demo of the problem**:
1- First we need a "problematic" script, which would contain a instance of the visited function. It could be a bigger script or a simple one with just one node. For this example we will be using this:

```
<<if not visited("Start")>>
    Red: first
    <<jump Start>>
<<endif>>
Red: second
===
```

2- Then we generate a new Yarn Project from the Unity button of the script itself.
![image](https://user-images.githubusercontent.com/5648170/170251285-d808db69-3f74-49fd-946b-a453c92db6c6.png)

3- After that we'll get the new Yarn Project which seems to work fine. Now we have to broke it. To do this we just have to apply any change to the Language form, like switching the base Language to a different one (no matter which). Then we apply the new settings.
![image](https://user-images.githubusercontent.com/5648170/170251382-c7add97a-2a99-4b77-bb1d-831131b8210b.png)
![image](https://user-images.githubusercontent.com/5648170/170251446-70c4005f-763d-43ce-a9ab-401cc700f283.png)

4- Now the Yarn Project is broken. 
Before:
![image](https://user-images.githubusercontent.com/5648170/170251939-ef561051-9364-4517-b373-dfb94b8e9335.png)
After:
![image](https://user-images.githubusercontent.com/5648170/170251781-b7755f22-818c-4214-88e7-58125976a8ea.png)
Some error messages appear on the console:
![image](https://user-images.githubusercontent.com/5648170/170251870-27b414a5-b9e9-4dea-b610-ef5d0ed481b3.png)

**What is the expected behavior?**
The Yarn Project should not break whenever I make changes to the Language properties within the Unity's Inspector.

**Please tell us about your environment:**
  
  - Yarn Spinner Version: 2.2.0
  - Unity Version: 2022.1.1f1

**Other information** 
Link to the original chat message in the Yarn Spinner Community's Discord Sever:
https://discord.com/channels/754171172693868585/754171643990900776/978968643402625084

**Comment by dukiodev** [comment]
Another user is having a similar issue [without using the visited function](https://discord.com/channels/754171172693868585/754171643990900776/978984774855192576):
https://discord.com/channels/754171172693868585/754171643990900776/978740315303346246

**Comment by desplesda** [comment]
Fixed in ac33e9168b40a64084215f2f42e5c28d0406874c. Thank you very much for the detailed report!
